via.go

```diff
@@ -0,0 +1,176 @@
+package kramer
+
+import (
+	"bufio"
+	"context"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"net"
+	"regexp"
+	"time"
+)
+
+const (
+	viaReboot = "Reboot"
+	viaReset  = "Reset"
+)
+
+// VIA Struct that defines general parameters needed for any VIA
+type Via struct {
+	Address  string
+	Username string
+	Password string
+	Logger   Logger
+}
+
+// These functions fulfill the DSP driver requirements
+// GetVolumeByBlock: opening a connection with the VIAs and then return the volume for the device
+func (v *Via) GetVolumeByBlock(ctx context.Context, block string) (int, error) {
+	return v.GetVolume(ctx)
+}
+
+// SetVolumeByBlock: Connect and set the volume on the VIA
+func (v *Via) SetVolumeByBlock(ctx context.Context, block string, volume int) (string, error) {
+	return v.SetVolume(ctx, volume)
+}
+
+// GetMutedByBlock: Return error because VIAs do not support a mute function
+func (v *Via) GetMutedByBlock(ctx context.Context, block string) (bool, error) {
+	return false, errors.New("Error in getting mute status of VIA, Feature not supported")
+}
+
+// SetMutedByBlock: Return error because VIAs do not support mute
+func (v *Via) SetMutedByBlock(ctx context.Context, block string, muted bool) error {
+	return errors.New("Error setting mute status of VIA, Feature not supported")
+}
+
+// GetInfo: needed by the DSP drivers implementation.  Will get hardware information
```

Comment:
Just double checking that we intentionally left this info blank? The comment made me think it should be getting info, but it's not. I'm fine with us not getting info if we aren't currently and don't have a reason to. If that's the case let's change the comment to mention that we don't get info, we're just fulfilling an interface. If we do want to get info "eventually", let's just throw in a TODO to remember that down the road.